.github/workflows/build.yml

```diff
@@ -0,0 +1,78 @@
+name: tardigrade terraform network mirror builder
+on:
+  # Run on demand
+  workflow_dispatch:
+
+  # Run pull requests against the main branch
+  pull_request:
+    branches: [main]
+    paths:
+      - 'Dockerfile.tools'
+      - '.github/workflows/dependabot_hack.yml'
+      - 'provider_versions.tf'
+
+  # Run when a release is created
+  release:
+    types: [released]
+
+permissions:
+  id-token: write
+
+concurrency:
+  group: ${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  BuildMirror:
+    name: build-mirror
+    runs-on: ubuntu-latest
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+      MIRROR_ENDPOINT: https://hashicorp-mirror.cloudarmor.io
+      MIRROR_BUCKET: p3-hashicorp-mirror
+      PROVIDERS_PREFIX: registry.terraform.io
+      REPO_PREFIX: repo
+      MIRROR_PATH: .mirror
+    steps:
+      - name: Clone this git repository
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+
+      - name: Install aws-cli
+        uses: unfor19/install-aws-cli-action@35a9630be0168293ad2afccbe06e8e9f47678d2c
+
+      - name: Download repo tools
+        run: |
+          make download-all
+
+      - name: configure aws credentials
+        if: github.event_name != 'pull_request'
+        uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/${{ secrets.AWS_ROLE_NAME }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+
+      - name:  Validate credential
+        if: github.event_name != 'pull_request'
+        run: aws sts get-caller-identity
+
+      - name: Push repo tools to s3 bucket
+        if: github.event_name != 'pull_request'
+        run: |
+          aws s3 cp ./${{ env.MIRROR_PATH }}/${{ env.REPO_PREFIX }}/ s3://${{ env.MIRROR_BUCKET }}/${{ env.REPO_PREFIX }}/
+
+      - name: Retrieve current providers mirror
+        if: github.event_name != 'pull_request'
```

Comment:
This command _shouldn't_ require credentials. So you _could_ pull it, if there's value to that.